src/modules/auth/auth.controller.ts

```diff
@@ -143,4 +148,54 @@ export class AuthController extends Controller {
       data: { message: "Successfully signedout from all devices" },
     });
   }
+
+  @Post("/forgotPassword")
+  @Middlewares(validateData(authValidations.forgotPass))
+  public async forgotPassword(
+    @Body() body: AuthVerifyEmail
+  ): Promise<APIResponse<Message>> {
+    const user = await userService.fetchByEmail(body.email);
+
+    if (!user) {
+      this.setStatus(statusConst.notFound.code);
+      return toResponse({ error: statusConst.notFound.message });
+    }
+
+    await otpService.send(user, "forgotPassword");
+
+    return toResponse({
+      data: { message: "Forgot password email send successfully" },
+    });
+  }
+
+  @Post("/resetPassword")
+  @Middlewares(authValidations.resetPass)
+  public async resetPassword(
+    @Body() body: AuthResetPass
+  ): Promise<APIResponse<Message>> {
+    const user = await userService.fetchByEmail(body.email);
+    if (!user) {
+      this.setStatus(statusConst.notFound.code);
+      return toResponse({ error: statusConst.notFound.message });
+    }
```

Comment:
We don't necessarily need this check here as the `otpService.verify` will implicitly verify the existence of user anyways